zhaquirks/inovelli/VZM31-SN.py

```diff
@@ -0,0 +1,108 @@
+"""VZM31-SN Two in One Switch/Dimmer Module."""
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.inovelli import (
+    INOVELLI_AUTOMATION_TRIGGERS,
+    Inovelli_VZM31SN_Cluster,
+)
+from zigpy.profiles import zha
+from zigpy.profiles.zha import DeviceType
+from zigpy.quirks import CustomDevice
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Identify,
+    LevelControl,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.smartenergy import Metering
+
+INOVELLI_VZM31SN_CLUSTER_ID = 64561
+
+
+class InovelliVZM31SN(CustomDevice):
+    """VZM31-SN 2 in 1 Switch/Dimmer Module."""
+
+    signature = {
+        MODELS_INFO: [("Inovelli", "VZM31-SN")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,  # 0
+                    Identify.cluster_id,  # 3
+                    Groups.cluster_id,  # 4
+                    Scenes.cluster_id,  # 5
+                    OnOff.cluster_id,  # 6
+                    LevelControl.cluster_id,  # 8
+                    Metering.cluster_id,  # 1794
+                    ElectricalMeasurement.cluster_id,  # 2820
+                    INOVELLI_VZM31SN_CLUSTER_ID,  # 64561
+                ],
+                OUTPUT_CLUSTERS: [Identify.cluster_id, Ota.cluster_id],  # 3  # 19
+            },
+            2: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
+                INPUT_CLUSTERS: [Identify.cluster_id],  # 3
+                OUTPUT_CLUSTERS: [
+                    Identify.cluster_id,  # 3
+                    OnOff.cluster_id,  # 6
+                    LevelControl.cluster_id,  # 8
+                    INOVELLI_VZM31SN_CLUSTER_ID,  # 64561
+                ],
+            },
+        },
+    }
+
+    replacement = {
+        MODELS_INFO: [("Inovelli", "VZM31-SN")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
```

Comment:
@Adminiuga should we map `DeviceType.DIMMER_SWITCH` to the light platform so that quirks don't have to change the device type like this?

Reply:
PLEASE do. In development of this device I had to override in the very beginning to even get any dim levels. All dimmers need 0-100% brightness_pct attribute, etc, as provided by the light domain.

Reply:
isn't `DeviceType.DIMMER_SWITCH` a "remote" device? Device really should report itself as "Dimmable_Light". 
Check Section 7.5.5 of Zigbee HA Profile, document 05-3520-29. DIMMER_SWITCH is mandated to have client on/off, level control and identity clusters. 
![image](https://user-images.githubusercontent.com/5826160/161875050-f2cb2798-fdbd-4c8e-a890-d0de3508a51f.png)

Reply:
@kreene1987 what kind of dimmer is it? two gang or single gang?

Reply:
> isn't `DeviceType.DIMMER_SWITCH` a "remote" device? Device really should report itself as "Dimmable_Light".
> Check Section 7.5.5 of Zigbee HA Profile, document [05-3520-29](tel:05-3520-29). DIMMER_SWITCH is mandated to have client on/off, level control and identity clusters.
> ![image](https://user-images.githubusercontent.com/5826160/161875050-f2cb2798-fdbd-4c8e-a890-d0de3508a51f.png)

Great point. I didn’t check this before hand.

Reply:
It is a single gang device. I might need a consult on this from @InovelliUSA but I believe we selected the class we did because this device has scene capabilities (multitap up to 5x) and the functionality of the dimmer_switch did not present those, although per the screenshot above it does seem that these are optional. @codyhackw perhaps we can try this offline and report back. We would need both scenes and groups to match the functions of the unit.

Reply:
If it is a single gang device, then the endpoint id 1 should be of a DIMMING_LIGHT device type, and the 2nd endpoint should be of a DIMMER_SWITCH device type. Similar to how GE46857 implemented it, none of the others got it right.

Reply:
> Similar to how GE46857 implemented it, none of the others got it right.

Can you link me to where this quirk is stored? I couldn't find GE or Jasco as a manuf in the quirks folder.

We'll get right on it for testing.

Reply:
Yes, I meant that the correct fix is to update the device firmware. There's no quirk for GE 45857, but you can find signature in https://github.com/home-assistant/core/blob/bc194cd209d4261bf7e15aa89eac09be5685c3db/tests/components/zha/zha_devices_list.py#L1674

Reply:
> Why not change the firmware of the device so that it reports itself as the correct device types? You can do it with a quirk but if you can change the firmware to be more accurate it may help other platforms as well.

Thanks for the feedback all, we will discuss with the engineer.